src/System.Device.Gpio/System/Device/Gpio/GpioController.Linux.cs

```diff
@@ -28,19 +28,37 @@ public GpioController(PinNumberingScheme numberingScheme)
         /// Attempt to get the best applicable driver for the board the program is executing on.
         /// </summary>
         /// <returns>A driver that works with the board the program is executing on.</returns>
-        private static GpioDriver GetBestDriverForBoard()
+        public static GpioDriver GetBestDriverForBoard()
         {
             string[] cpuInfoLines = File.ReadAllLines(CpuInfoPath);
-            Regex regex = new Regex(@"Hardware\s*:\s*(.*)");
-            foreach (string cpuInfoLine in cpuInfoLines)
+            Regex regexHw = new Regex(@"Hardware\s*:\s*(.*)");
+            Regex regexRev = new Regex(@"Revision\s*:\s*(.*)");
```

Comment:
We are trying to move away from using Regex, which is not performant and can sometimes be unsafe to use. Is there a way to get the fact that we are running on a Pi4 via a different mechanism?

Reply:
see @Frankenslag comment above. He has been working on getting new hardware detection logic which is moving away from Regex.

Reply:
Yes, I know. I'm actually waiting for that branch to be merged so this can be combined. However, I think that we need a more sophisticated concept for detecting compatibility between drivers and hardware.